### DISCUSSION TITLE
Release Candidate 2.0.0

### DISCUSSION THREAD

**Discussion by rzo1** [dkpro/dkpro-jwpl#281] [Announcements]
Hi folks,

a new version of JWPL is in the starting blocks. 

This version contains many improvements and bug fixes, e.g. the new release supports the Jakarta namespace and sets the Java language level to Java 17. 
Please note, that the packages and artifact ids also changed, so this release is considered breaking.

An overview of the changes can be found at https://github.com/dkpro/dkpro-jwpl/milestone/14?closed=1

This version should work with the Wikipedia dumps from 2022. Since there was also a modernization of the Mediawiki schema, we will bring the corresponding support to Swebble in a next step. Unfortunately, development has also fallen asleep here, so we have to do a bit more ;-) But help and PRs are always welcome!

Here is the staging repository:

https://oss.sonatype.org/content/repositories/orgdkprojwpl-1003/

Here is the tag: 

https://github.com/dkpro/dkpro-jwpl/releases/tag/dkpro-jwpl-2.0.0

Please provide feedback via GitHub discussions or on the related mail thread.

Best regards
Richard

**Comment by mawiesne** [comment]
+1 to publish this release candidate.

**Comment by mawiesne** [comment]
FYI @tgalery

**Comment by reckart** [comment]
I  locally upgraded the `dkpro-core-io-jwpl-asl` code to build against the RC. Looks like the code compiles. Unfortunately, the unit tests do not work anymore since the test DB is no longer available :(

**Comment by reckart** [comment]
I'm still planning to have a look at the artifacts themselves, but didn't get to it yet...

**Comment by reckart** [comment]
* ✅ Upgraded DKPro Core JWPL Reader against the RC - looks good
* ⚠️ Notice file in util module contains a strange class name: `parser.templates.util.org.dkpro.jwpl.SectionExtractor:`
* ⚠️ swc-engine-shade module does not remap the packages - this can lead to problems in scenarios where JWPL is used together with another library that uses a non-shaded sweble. Can we remap the packages?
* ⚠️ swc-engine-shade NOTICE file does not describe the origin and original version of the artifact
* ⚠️ revision-machine module contains a "package" called `configSamples` - does this need to be part of the JAR?
* ⚠️ jwpl-api module contains a packaged called `org.dkpro.jwpl.util` - there should only be a single top-level package in each JAR - this package overlaps with a package of the same name in the jwpl-util module. Can the package in jwpl-api be renamed or moved to jwpl-util?
* ⚠️ the jwpl-api module contains a file `jwpl-env-configuration.properties` in the root packages. The root package should not be used for files or classes. Is the file needed? Can it be moved to a proper package?
* ⚠️ the jwpl-api module contains a `log4j2.xml` file - such files may be acceptable in far-jar modules that are not used as libraries by other application - but the jwpl-api module seems to be a library module and thus should not contain such a file.
* ⚠️ the jwpl-util module contains files `jwpl-env-configuration.properties` in the root packages. Are they needed? Can they be moved to a proper package?

**Reply by rzo1** [reply]
>  Notice file in util module contains a strange class name: parser.templates.util.org.dkpro.jwpl.SectionExtractor:

Fixed.

> swc-engine-shade NOTICE file does not describe the origin and original version of the artifact

Fixed.

> swc-engine-shade module does not remap the packages - this can lead to problems in scenarios where JWPL is used together with another library that uses a non-shaded sweble. Can we remap the packages?

Added a related relocation.

> revision-machine module contains a "package" called configSamples - does this need to be part of the JAR?

I don't care about these files being part of the JAR (as they were part of it for ages now). I moved them to the README.md of the related module.

>  jwpl-api module contains a packaged called org.dkpro.jwpl.util - there should only be a single top-level package in each JAR - this package overlaps with a package of the same name in the jwpl-util module. Can the package in jwpl-api be renamed or moved to jwpl-util?

No, we cannot merge them as `dkpro-jwpl-util` is a somehow joined module between the revision and jwpl api. I will move them to "api.util" instead.

> the jwpl-api module contains a file jwpl-env-configuration.properties in the root packages. The root package should not be used for files or classes. Is the file needed? Can it be moved to a proper package?

Same as below. Misconfiguration, so it was leaking from the integration test.

>  the jwpl-api module contains a log4j2.xml file - such files may be acceptable in far-jar modules that are not used as libraries by other application - but the jwpl-api module seems to be a library module and thus should not contain such a file.

The `log4j2.xml` file is only contained in the `src/test/resources/` or `src/it/resources` folder, which is ok. However, the build process includes the integration test log4j2.xml into the resulting file, which is NOT ok. Fixed the misconfiguration + removed the unneeded exec JAR generation (no executable classes within API)


> the jwpl-util module contains files jwpl-env-configuration.properties in the root packages. Are they needed? Can they be moved to a proper package?

It was leaking from the API module due to a fat jar generation (to have an executable + the resource filtering issue mentioned above), which isn't needed for the API utils (as there are no main classes).

**Reply by rzo1** [reply]
https://github.com/dkpro/dkpro-jwpl/pull/283

**Reply by rzo1** [reply]
@reckart @mawiesne I re-rolled the release with the fixes from #283 and updated the announcement to point to the new staging repository. Please check ;-)

**Reply by reckart** [reply]
I locally checked out the 2.0.0 tag and built.

I can still see the `inflating: create_tpltables_simplegenerator.sql` in the dkpro-jwpl-util-2.0.0.jar

At least DKPro Core is using `dkpro-jwpl-revisionmachine-2.0.0.jar` as a library which means that it is running into a problem with the log4j2.xml file contained in that JAR. It seems not to be a fat JAR, so maybe having that file in there is not necessary?

There are tons of warnings in the build log from the build.

That said... I wouldn't -1 a release either at this point. So +0 from me. There is still some stuff that could be improved (in particular the log4j2.xml)... So +0 from me.

**Reply by rzo1** [reply]
@reckart See https://github.com/dkpro/dkpro-jwpl/pull/285 - this should remove log4j2 from the jars distributed to mvn central, moves the sql files to README and fixes some shade issues.

There are a lot of Javadoc warnings / issues inside the build log,  but that is something for the next minor version of JWPL (imho). The Javadoc plugin seems to be configured really harsh forcing to document even useless things ;-)

**Reply by rzo1** [reply]
After #285 is included, I will build another RC and we should be good to go (for now), imho.

**Comment by mawiesne** [comment]
+1
